---
title: "16S rRNA proscessing"
author: "Ioannis D. Kampouris"
date: "2023-21-06"
output: pdf_document
---
#############################################################
Code in Bash Shell
#############################################################
#!/bin/bash
#Author: Ioannis Kampouris
#Purpose: Use fastqc and multiqc in the R Server (UBUNTU OS)
#Dependencies a path for fastqc and 
# Attention: To download multiQC you need to have a git clone with pip3 install .

export PATH=$PATH:"/home/ioannis.kampouris/fastqc/FastQC" 
#Change accordingly to your path 
export PATH=$PATH:"/home/ioannis.kampouris/fastqc/MultiQC" 
#Change accordingly to your path 

find -name  "*raw_1*"|sed 's/^..//'  > sample_list.txt 
#Change accordingly to your path 

mkdir QC
filename='sample_list.txt'
echo Start
while read i;
   do 
   SAMPLE=$(echo ${i} | sed "s/raw_1\.fq\.gz//")
      echo "$SAMPLE"
   fastqc  ${SAMPLE}raw_2.fq.gz -o QC #You can remove the 
   #"RAW" if you want your files with NOVOGENE cleaning
    fastqc  ${SAMPLE}raw_1.fq.gz -o QC
     
done <"$filename"
 
 mkdir QC/FW
 mkdir QC/RE

mv QC/*raw_1* QC/FW
mv QC/*raw_2* QC/RE
cd QC/FW
python3 -m multiqc  .
cd ..
cd ..
cd QC/RE
python3 -m multiqc  .

After I run the fastqc/multiqc I plotted the results of average quality per read
##########################################################################
```{r, fig.height=10, fig.width=10}
library(readr)
library(tidyverse)
library(ggplot2)

set.seed(17011990)
```
Average quality Above Phred_Score 30 for most of the samples, I can continue.

```{r, fig.height=10, fig.width=10}
quality_FW_2020=read_delim(paste0(
"/home/ioannis.kampouris/QC/FW/",
"multiqc_data/mqc_fastqc_per_base_sequence_quality_plot_1.txt"), delim = "\t")%>%
  gather(-Sample, key="Sequence", value = "Average_Phred_Score") %>%
  mutate(Type="Forward")

quality_RE_2020=read_delim(paste0(
"/home/ioannis.kampouris/",
"QC/RE/multiqc_data/mqc_fastqc_per_base_sequence_quality_plot_1.txt"), 
delim = "\t")%>%
gather(-Sample, key="Sequence", value = "Average_Phred_Score")%>%
mutate(Type="Reverse")
QC_both_FW_and_RE_2020=full_join(quality_FW_2020, quality_RE_2020, 
by=colnames(quality_FW_2020))

ggplot(QC_both_FW_and_RE_2020, aes(y=Average_Phred_Score,
x=as.numeric(Sequence), colour=Sample))+
facet_grid(~Type)+ geom_line() + 
geom_hline(yintercept = 25, linetype="dashed")+
theme_bw() + theme( axis.text.y = 
element_text( size=25, face = "bold", colour="black")) +
theme(axis.text.x = 
element_text(size = 25, face = "bold", colour="black")) +
theme(legend.text = element_text( size=25, face="bold", colour = "black")) + 
theme( axis.title.y = element_text( size=25, face = "bold", colour="black")) + 
theme(axis.title.x = element_text(size = 25, face = "bold", colour="black")) +
theme(legend.title =  element_text( size=25, face="bold", colour = "black"))+
theme(strip.background =  element_rect( fill = "#222d80",  colour = "#222d80"))+
theme(strip.text  =  element_text( size=10, face="bold", colour = "white")) +
theme(legend.position = "none") +
ggtitle("Raw Sequences quality 2020") + xlab("Base Number") + ylim(20, 40)

```  
#Start dada2 for the 2020 samples
```{r, fig.height=10, fig.width=10}
library(dada2)
path_2020 <-paste0( "/home/ioannis.kampouris/",
"trimmed_cutadapt")
# CHANGE ME to the directory containing your demultiplexed fastq files
filtpathF_2020 <- file.path(path_2020, "FW_filtered") 
# Filtered forward files go into the pathF/filtered/ subdirectory
filtpathR_2020 <- file.path(path_2020, "RE_filtered") 
fastqFs_2020 <- sort(list.files(path_2020, pattern="_1.fq"))
fastqRs_2020 <- sort(list.files(path_2020, pattern="_2.fq"))
# Filtering
out_2020<-filterAndTrim(fwd=file.path(path_2020, fastqFs_2020), filt=file.path(filtpathF_2020, fastqFs_2020),
              rev=file.path(path_2020, fastqRs_2020), filt.rev=file.path(filtpathR_2020, fastqRs_2020),
               maxEE=c(2,2), rm.phix=TRUE, matchIDs = T,
              trimRight =   c(5,5),
              compress=TRUE, verbose=TRUE, multithread=TRUE)

```

Lowest output ~75.9%
```{r, fig.height=10, fig.width=10}
filtFs_2020 <- list.files(filtpathF_2020, pattern="fq", full.names = TRUE)
filtRs_2020 <- list.files(filtpathR_2020, pattern="fq", full.names = TRUE)
sample.names_2020 <- sapply(strsplit(basename(filtFs_2020), "_"), `[`, 1) # Assumes filename = samplename_XXX.fastq.gz
sample.namesR_2020 <- sapply(strsplit(basename(filtRs_2020), "_"), `[`, 1) # Assumes filename = samplename_XXX.fastq.gz
if(!identical(sample.names_2020, sample.namesR_2020)) stop("Forward and reverse files do not match.")else{"Proceed with Safety"}
names(filtFs_2020) <- sample.names_2020
names(filtRs_2020) <- sample.names_2020

```
```{r}
source("mod_loess_function.R")
errF_2020 <- learnErrors(filtFs_2020, nbases=1e8,randomize = T, multithread=TRUE)
errR_2020 <- learnErrors(filtRs_2020, nbases=1e8,randomize = T, multithread=TRUE)
plotErrors(errF_2020, nominalQ=TRUE)
plotErrors(errR_2020, nominalQ=TRUE)
errF_2020_mod <- learnErrors(filtFs_2020,  errorEstimationFunction = loessErrfun_mod1, verbose = T, randomize = T, nbases=1e8, multithread=TRUE)
errR_2020_mod <- learnErrors(filtRs_2020, nbases=1e8,  randomize = T, verbose = T, errorEstimationFunction = loessErrfun_mod1, multithread=TRUE)
plotErrors(errF_2020_mod, nominalQ=TRUE)
plotErrors(errR_2020_mod, nominalQ=TRUE)

```

```{r}
ddF_2020 <- dada(filtFs_2020, err=errF_2020_mod, multithread=TRUE)
ddR_2020 <- dada(filtRs_2020, err=errR_2020_mod, multithread=TRUE)
```
```{r}
mergers_2020 <- mergePairs(ddF_2020, filtFs_2020, ddR_2020, filtRs_2020, minOverlap=7)
getN <- function(x) sum(getUniques(x))
track_2020 <- cbind(out_2020, sapply(ddF_2020, getN), sapply(ddR_2020, getN), sapply(mergers_2020, getN))
```
```{r}
# Construct sequence table and remove chimeras
seqtab_2020.nochim <- makeSequenceTable(mergers_2020)
```


```{r,ig.height=10, fig.width=10}
table(nchar(getSequences(seqtab_2020.nochim)))

seqtab2 <- seqtab_2020.nochim[,nchar(colnames(seqtab_2020.nochim)) %in% 403:431]
getN <- function(x) sum(getUniques(x))
cbind(out_2020, sapply(ddF_2020, getN), sapply(ddR_2020, getN), sapply(mergers_2020, getN))
#Very large amount of sequences merged.
seqtab.afterchim <- removeBimeraDenovo(seqtab2, method="consensus", multithread=TRUE)
track= cbind(out_2020, sapply(ddF_2020, getN), sapply(ddR_2020, getN), sapply(mergers_2020, getN),rowSums(seqtab2),rowSums(seqtab.afterchim))
colnames(track)<-c("Reads In", "Reads Out", "ddR", "ddF","merged", "size_out","Chimera_out")
track2=track/track[,1]

ggplot(as.data.frame(track),aes(x=((1-Chimera_out/size_out))))+ 
  geom_histogram()+ ylab("Density")+xlab("Chimeric ratio of Reads")+
  ggtitle("Distribution of Chimeric/Merged reads proportion")

```


```{r}


ASVtable=as.data.frame(t(seqtab.afterchim))%>%rownames_to_column(var="Sequence")
taxa <- assignTaxonomy(seqtab.afterchim, 
"~/silva_naive/silva_nr99_v138.1_train_set.fa.gz", minBoot = 80, multithread=TRUE)

complete_ASV_table= full_join(as.data.frame(taxa)%>%rownames_to_column(var = "Sequence"), ASVtable, by="Sequence")
complete_ASV_table$ASV=paste0("ASV", rownames(complete_ASV_table))

clean_ASV_table = complete_ASV_table%>%
  filter(Kingdom!="NA"&Order!="Chloroplast"&Family!="Mitochondria")%>%select(Sequence, ASV, everything())
write.csv(complete_ASV_table,"dada2_out/ASV_Table_LTE_2020.csv")
write.csv(clean_ASV_table,"dada2_out/plastid_clean_ASV_Table_LTE_2020.csv")
track_final= cbind(track,colSums(clean_ASV_table[,9:ncol(clean_ASV_table)]) )
track_final_ratio= cbind(track,colSums(clean_ASV_table[,9:ncol(clean_ASV_table)]) )/track[,1]

writeFasta<-function(data, filename){
  fastaLines = c()
  for (rowNum in 1:nrow(data)){
    fastaLines = c(fastaLines, 
as.character(paste(">", data[rowNum,"ASV"], sep = "")))
fastaLines = c(fastaLines,as.character(data[rowNum,"Sequence"]))
  }
  
  fileConn<-file(filename)
  writeLines(fastaLines, fileConn)
  close(fileConn)
}

writeFasta(clean_ASV_table%>%select(ASV, Sequence), "dada2_out/ASVs_LTE_2020.fa")

```
