
```{r}
ASVtable_working <- read.csv("~/plastid_clean_low_read_clean_contaminant_free_ASV_Table_LTE_2020.csv", row.names = 1)
gathered_ASV_table=gather(ASVtable_working[c(1,3:114)], -ASV, key = "Sample", value = "Abundance")
#Assign metadata via sample names
metadata_assign=data.frame(Sample=unique(gathered_ASV_table$Sample))%>%
  mutate(Category=Sample)%>%
  mutate(Category=str_replace(Category, "FSRW", "Root-Associated_Soil/Root-Window;"))%>%
  mutate(Category=str_replace(Category, "RHRW", "Rizosphere/Root-Window;"))%>%
  mutate(Category=str_replace(Category, "RW", "Root-Window;"))%>%
  mutate(Category=str_replace(Category, "FS", "Root-Associated_Soil;"))%>%
  mutate(Category=str_replace(Category, "RH", "Rhizosphere;"))%>%
  mutate(Category=str_replace(Category, ".T0", "T0-Soil;"))%>%
   mutate(., Category=str_replace(Category, ".raw", ""))%>%
  separate(., Category, into = c("Type", "Rest"), sep=";")%>%
mutate(., Rest=str_replace(Rest, ".CT.Ext.C", "CT;Extensive;Ctrl;"))%>%
  mutate(., Rest=str_replace(Rest, ".CT.Ext.I", "CT;Extensive;BMc;"))%>%
       mutate(., Rest=str_replace(Rest, ".CTEx.I", "CT;Extensive;BMc;")) %>%
       mutate(., Rest=str_replace(Rest, ".CTEx.C", "CT;Extensive;Ctrl;")) %>%
       mutate(., Rest=str_replace(Rest, ".CTIn.I", "CT;Intensive;BMc;")) %>%
       mutate(., Rest=str_replace(Rest, ".CTIn.C", "CT;Intensive;Ctrl;")) %>%
       mutate(., Rest=str_replace(Rest, ".CTIn.I", "CT;Intensive;BMc;")) %>%
       mutate(., Rest=str_replace(Rest, ".CTIn.C", "CT;Intensive;Ctrl;")) %>%
       mutate(., Rest=str_replace(Rest, ".CTIn.C", "CT;Intensive;Ctrl;")) %>%
       mutate(., Rest=str_replace(Rest, ".CTIn.C", "CT;Intensive;Ctrl;")) %>%
       mutate(., Rest=str_replace(Rest, ".CT.Int.C", "CT;Intensive;Ctrl;")) %>%
       mutate(., Rest=str_replace(Rest, ".CT.Int.I", "CT;Intensive;BMc;")) %>%
       mutate(., Rest=str_replace(Rest, ".CTEx.C", "CT;Extensive;Ctrl;")) %>%
       mutate(., Rest=str_replace(Rest, ".CT.Int.I", "CT;Intensive;BMc;")) %>%
       mutate(., Rest=str_replace(Rest, "CTExtT0-Soil.", "CT;Extensive;T0-Soil;")) %>%
       mutate(., Rest=str_replace(Rest, "CTIntT0-Soil.", "CT;Intensive;T0-Soil;")) %>%
       mutate(., Rest=str_replace(Rest, "MPExtT0-Soil.", "MP;Extensive;T0-Soil;")) %>%
       mutate(., Rest=str_replace(Rest, "MPIntT0-Soil.", "MP;Intensive;T0-Soil;")) %>%
       mutate(., Rest=str_replace(Rest, ".MP.Ext.C", "MP;Extensive;Ctrl;")) %>%
       mutate(., Rest=str_replace(Rest, ".MP.Ext.I", "MP;Extensive;BMc;")) %>%
       mutate(., Rest=str_replace(Rest, ".MPEx.I", "MP'Extensive;BMc;")) %>%
       mutate(., Rest=str_replace(Rest, ".MPEx.C1", "MP;Extensive;Ctrl;")) %>%
       mutate(., Rest=str_replace(Rest, ".MPIn.I", "MP;Intensive;BMc;")) %>%
       mutate(., Rest=str_replace(Rest, ".MPIn.C", "MP;Intensive;Ctrl;")) %>%
       mutate(., Rest=str_replace(Rest, ".MPIn.I", "MP;Intensive;BMc;")) %>%
       mutate(., Rest=str_replace(Rest, ".MPIn.C", "MP;Intensive;Ctrl;")) %>%
       mutate(., Rest=str_replace(Rest, ".MPIn.C", "MP;Intensive;Ctrl;")) %>%
       mutate(., Rest=str_replace(Rest, ".MPIn.C", "MP;Intensive;Ctrl;")) %>%
       mutate(., Rest=str_replace(Rest, ".MP.Int.C", "MP;Intensive;Ctrl;")) %>%
       mutate(., Rest=str_replace(Rest, ".MP.Int.I", "MP;Intensive;BMc;")) %>%
  separate(., Rest, into=(c("Practice1", "N-Fertilization_Intensity", "Treatment", "Block")), sep = ";")%>% 
    filter(., Practice1=="CT")
RS_metadata=filter(metadata_assign, Type=="Rhizosphere")
rownames(ASVtable_working)=ASVtable_working$ASV
ASVtable1=( ASVtable_working[,RS_metadata$Sample])
ASVtable2=apply(ASVtable1, 2, function(x) 100*(x)/sum(x))
betadiv16S=vegdist(log10(t(ASVtable2)+1), method = "bray")
betadiv16S2=cmdscale(betadiv16S, eig = T, k=2)
betadiv16S3=scores(betadiv16S2, display="sites")
betadiv16S4=cbind(betadiv16S3[RS_metadata$Sample,],RS_metadata)
betadiv16S4$Treatment2=factor(betadiv16S4$Treatment, levels = c(
"Ctrl",   "BMc"
))
beta_ASVs= ggplot(betadiv16S4, 
       aes(x=Dim1, y=Dim2, colour=Treatment2,  
           ))+
  geom_point(size=5,aes(shape=`N-Fertilization_Intensity`)) + 
  theme_bw() + 
  scale_colour_manual(name="",values=c("#146eb4", "#ff9900"))+
  theme(strip.background = element_rect(colour="white", fill="white"))+
  theme( axis.text.y = element_text( size=15, face = "bold", colour="black")) +
  theme(axis.text.x = element_text(size = 15, face = "bold", colour="black", angle = 0)) +
  theme(legend.text = element_text( size=25, face="bold", colour = "black")) + 
  theme( axis.title.y = element_text( size=15, face = "bold", colour="black")) + 
  theme(axis.title.x = element_text(size = 15, face = "bold", colour="black")) +
  theme(legend.title =  element_text( size=25, face="bold", colour = "black")) +
  #guides(shape=guide_legend(override.aes = list(size=22), ncol=1)) + 
  guides(size=guide_legend( ncol=1)) + #geom_errorbar(stat="summary",position = position_dodge2(width = 1))+
ylab(paste0("MDS2[", round(100*betadiv16S2$eig[2]/sum(betadiv16S2$eig),2),"%]")
                    )+
  xlab(paste0("MDS1[", round(100*betadiv16S2$eig[1]/sum(betadiv16S2$eig),2),"%]")
                    )+stat_ellipse(aes(group = ))


permanova1= adonis(betadiv16S~`N-Fertilization_Intensity`*Treatment,
       RS_metadata, permutations = 9999, by="terms")
```


```{r}
fungi_RA=   read.table(paste0("~/otu_RH.fungi.Ioannis.txt"), sep=" ")%>%t()

fungi_RA=apply(fungi_RA, 1, function(x) 100*(x)/sum(x))%>%as.data.frame()%>%
  rownames_to_column(var="Col")
fungi_RA$OTU=paste0("OTU", row.names(fungi_RA))

OTU_Table_count1=read.table(paste0("~/otu_RH.fungi.Ioannis.txt"), sep=" ")%>%
  rownames_to_column(var="Col") %>%full_join(select(fungi_RA, Col, OTU), by="Col")%>%
  select(-Col)%>%column_to_rownames(var="OTU")%>%t()%>%
  as.data.frame()
fungi_taxonomy=   read.table(paste0("~/ITS2_CT_2020_TAX_reduced.txt"), sep="\t")%>%

  rownames_to_column(var="Col")%>%full_join(fungi_RA%>%select(Col, OTU))%>%filter(
    OTU!="NA"
  )





fungi_taxonomy[fungi_taxonomy=="NA"]="Unclassified"
fungi_taxonomy[fungi_taxonomy=="uncultured"]<-"Unclassified"
fungi_taxonomy[fungi_taxonomy=="metagenome"]<-"Unclassified"

gathered_fungi_table=gather(fungi_RA, -OTU, key = "Sample", value = "Abundance")

#Assign metadata via sample names
metadata_assign=data.frame(Sample=unique(gathered_fungi_table$Sample))%>%
  mutate(Category=Sample)%>% 
  separate(., Category, into=(c("Type", "Practice1", "N-Fertilization_Intensity", "Treatment")), sep = "_")%>% 
  filter(., Practice1=="CT")

metadata_assign$Block=substr(metadata_assign$Treatment,nchar(metadata_assign$Treatment), nchar(metadata_assign$Treatment)+1)
metadata_assign$Treatment2=substr(metadata_assign$Treatment,1, nchar(metadata_assign$Treatment)-1)


metadata_assign$Treatment=factor(metadata_assign$Treatment2, c("Ctrl","BMc"))
RS_metadata=filter(metadata_assign, Type=="RH")
fungi_taxonomy2=mutate(fungi_taxonomy%>%select(-Col), Genus=paste0(phylum,";",class,";",order,";",family,";",genus))%>%
  mutate(., Family=paste0(phylum,";",class,";",order,";",family))%>%
  mutate(., Order=paste0(phylum,";",class,";",order))%>%
  mutate(., Class=paste0(phylum,";",class))%>%mutate(., OTU2=paste0(genus,";",OTU))%>%
  mutate(Phylum=phylum)



input1=fungi_RA%>% select(-Col)%>%column_to_rownames(var="OTU")


betadivITS=vegdist(log10(t(input1)+1), method = "bray")
betadivITS2=cmdscale(betadivITS, eig = T, k=2)
betadivITS3=scores(betadivITS2, display="sites")
betadivITS4=cbind(betadivITS3[RS_metadata$Sample,],RS_metadata)
betadivITS4$Treatment2=gsub("BM","BMc",betadivITS4$Treatment2)
betadivITS4$Treatment2=factor(betadivITS4$Treatment2, levels = c(
"Ctrl",   "BMc"
))
beta_ITS= ggplot(betadivITS4, 
       aes(x=Dim1, y=Dim2, colour=Treatment2,  
           ))+
  geom_point(size=5,aes(shape=paste0(`N-Fertilization_Intensity`,"ensive"))) + 
  theme_bw() + 
  scale_shape(name = "N-Fertilization_Intensity")+
  scale_colour_manual(name="",values=c("#146eb4", "#ff9900"))+
  theme(strip.background = element_rect(colour="white", fill="white"))+
  theme( axis.text.y = element_text( size=15, face = "bold", colour="black")) +
  theme(axis.text.x = element_text(size = 15, face = "bold", colour="black", angle = 0)) +
  theme(legend.text = element_text( size=25, face="bold", colour = "black")) + 
  theme( axis.title.y = element_text( size=15, face = "bold", colour="black")) + 
  theme(axis.title.x = element_text(size = 15, face = "bold", colour="black")) +
  theme(legend.title =  element_text( size=25, face="bold", colour = "black")) +
  #guides(shape=guide_legend(override.aes = list(size=22), ncol=1)) + 
  guides(size=guide_legend( ncol=1)) + #geom_errorbar(stat="summary",position = position_dodge2(width = 1))+
ylab(paste0("MDS2[", round(100*betadivITS2$eig[2]/sum(betadivITS2$eig),2),"%]")
                    )+
  xlab(paste0("MDS1[", round(100*betadivITS2$eig[1]/sum(betadivITS2$eig),2),"%]")
                    )+stat_ellipse(aes(group = ))

```

